### DISCUSSION TITLE
Task 3 - forward() got an unexpected keyword argument 'labels'

### DISCUSSION THREAD

**Discussion by DezignDezire** [sebastian-hofstaetter/teaching#76] [Advanced IR: Exercise Q&A] [answered]
Hello,

for task 3 we are using `AutoModelForQuestionAnswering` from the transformer library to load `deepset/bert-base-cased-squad2`. After preprocessing the data and creating a `Dataset` instance for the test and validation set respectively, we use the `Trainer.train()` for fine tuning. However, we get the following error: 

`forward() got an unexpected keyword argument 'labels'`

This comes as a surprise since this model should already be fine tuned to Squad and we want to explicitly use if for QA (and not for Sequence Classification, which uses the `labels` parameter in its `forward()` call).

The shape of our training set looks like following:

```
Dataset({
    features: ['input_ids', 'attention_mask', 'start_positions', 'end_positions'],
    num_rows: 11155
})
```

Please help 🏳
LG Phil

**Comment by DezignDezire** [comment, marked as answer]
Update: when using a custom training loop with loss and custom optimizer (instead of `Trainer.train()`) the fine tuning works fine 🤗
